### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by mpashka**
Log redis executable output
Add log start timeout
Make start message check more flexible (adopted for different redis versions)

### CODE REVIEW

**Comment by fmonniot** [comment]
Hey, thank you for the pull request.
I have looked over it really quickly (I'll take a look during the week) and I like the concept of State you have introduced :)
Just a quick question though, is it possible to add some test to not have the code coverage decreasing ?

Again, thanks to have taken the time to put this PR together.

**Comment by mpashka** [comment]
Sure, will add more tests to improve coverage

**Comment by mpashka** [comment]
Done

File: src/main/java/redis/embedded/AbstractRedisInstance.java [hunk 1]
```diff
@@ -47,42 +69,104 @@ public InputStream errors() {
...
             Thread.sleep(200);
 
             String outputLine;
-            do {
+            while (true) {
                 outputLine = reader.readLine();
                 if (outputLine == null) {
                     //Something goes wrong. Stream is ended before server was activated.
+                    //Set state to failed
+                    correctState: {
```
Comment (fmonniot) [inline src/main/java/redis/embedded/AbstractRedisInstance.java]: Why the use of a labelled statement here ?

File: src/main/java/redis/embedded/AbstractRedisInstance.java [hunk 2]
```diff
@@ -98,4 +182,19 @@ private void tryWaitFor() {
...
+
+    void setStartupTimeoutMs(long startupTimeoutMs) {
+        this.startupTimeoutMs = startupTimeoutMs;
+    }
+
+    State getState() {
+        return state.get();
+    }
+
+    enum State {
```
Comment (fmonniot) [inline src/main/java/redis/embedded/AbstractRedisInstance.java]: Could you move the enum into its own file ?
And add the associated `EnumSet` to it too.

File: src/main/java/redis/embedded/RedisSentinel.java [hunk 3]
```diff
@@ -9,9 +9,13 @@
 import java.nio.charset.Charset;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Pattern;
 
 public class RedisSentinel extends AbstractRedisInstance {
-    private static final String REDIS_READY_PATTERN = ".*Sentinel runid is.*";
+    private static final Pattern REDIS_READY_PATTERN = Pattern.compile(
```
Comment (fmonniot) [inline src/main/java/redis/embedded/RedisSentinel.java:15]: 👍

**Comment by fmonniot** [comment]
Hey there,
First sorry for having taking so long to make the review (and nitpitting in it, they was nothing much to say about it ;)).
Anyway, I can make the change to it if you don't have time for it.

Also, I'll add you to the contributors in the README.

**Comment by mpashka** [comment]
Created separate state object

**Review by fmonniot — APPROVED** [review]
Good Job 👍 
As soon as you have fixed the missing `!` I'm going to merge your changes.
Thanks for this great contribution !

File: src/main/java/redis/embedded/AbstractRedisInstance.java [hunk 4]
```diff
@@ -47,42 +60,99 @@ public InputStream errors() {
     }
 
     private void awaitRedisServerReady() throws IOException, InterruptedException {
+        if (state.isActivating()) {
```
Comment (fmonniot) [inline src/main/java/redis/embedded/AbstractRedisInstance.java]: I believe here it should be `if (!state.isActivating()) {` ?
I think it's why the tests are failing.

**Comment by mpashka** [comment]
@fmonniot Can we proceed with this?

**Comment by fmonniot** [comment]
Oh sorry, I haven't been notified by GitHub that you made the change :cry:

**Pull request merged**